src/nonce-manager/nonce-manager.ts

```diff
@@ -0,0 +1,50 @@
+import Contract from 'web3-eth-contract'
+import {AbiItem} from 'web3-utils'
+import {BlockchainProviderConnector} from '../connector'
+import {NonceRequestParams, OrderNonce} from './types'
+import LimitOrderV3ABI from '../abi/AggregationRouterV5.abi.json'
+import {decodeNonce, encodeNonce} from './utils'
+
+export class NonceManager {
+    public readonly makerAddress: string
+
+    public readonly provider: BlockchainProviderConnector
+
+    public readonly limitOrderV3Contract: Contract.Contract
+
+    constructor(params: NonceRequestParams) {
+        this.makerAddress = params.maker
+        this.provider = params.blockchainProvider
+
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
+        this.limitOrderV3Contract = new Contract(LimitOrderV3ABI as AbiItem[])
+    }
+
+    static new(params: NonceRequestParams): NonceManager {
+        return new NonceManager(params)
+    }
+
+    /**
+     * @param nonceRequestType enum OrderNonce, either Auto or Empty, in case of Empty will return undefined
+     */
+    async getNonce(
+        nonceRequestType: OrderNonce = OrderNonce.Auto
+    ): Promise<string | undefined> {
+        if (nonceRequestType === OrderNonce.Empty) {
+            return
+        }
```

Comment:
Don't need to use `nonceRequestType` here. In case user want to use this manager, he doesn't expect to use empty nonce

Reply:
Pass here `maker` address and don't use it in `NonceRequestParams`